pkg/controllers/management/k3sbasedupgrade/k3sbased_upgrade_handler.go

```diff
@@ -19,41 +15,36 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
-func (h *handler) onClusterChange(key string, cluster *v3.Cluster) (*v3.Cluster, error) {
+func (h *handler) onClusterChange(key string, cluster *apimgmtv3.Cluster) (*apimgmtv3.Cluster, error) {
 	if cluster == nil || cluster.DeletionTimestamp != nil {
 		return nil, nil
 	}
-	isK3s := cluster.Status.Driver == v32.ClusterDriverK3s
-	isRke2 := cluster.Status.Driver == v32.ClusterDriverRke2
-	// only applies to k3s/rke2 clusters
-	if !isK3s && !isRke2 {
-		return cluster, nil
-	}
-	// Don't allow nil configs to continue for given cluster type
-	if (isK3s && cluster.Spec.K3sConfig == nil) || (isRke2 && cluster.Spec.Rke2Config == nil) {
-		return cluster, nil
-	}
 
 	var (
 		updateVersion string
-		strategy      v32.ClusterUpgradeStrategy
+		strategy      apimgmtv3.ClusterUpgradeStrategy
 	)
-	switch {
-	case isK3s:
+
+	// only applies to imported k3s/rke2 clusters
+	if cluster.Status.Driver == apimgmtv3.ClusterDriverK3s {
+		if cluster.Spec.K3sConfig == nil {
+			return cluster, nil
+		}
 		updateVersion = cluster.Spec.K3sConfig.Version
 		strategy = cluster.Spec.K3sConfig.ClusterUpgradeStrategy
-	case isRke2:
+	} else if cluster.Status.Driver == apimgmtv3.ClusterDriverRke2 {
+		if cluster.Spec.Rke2Config == nil {
+			return cluster, nil
+		}
 		updateVersion = cluster.Spec.Rke2Config.Version
 		strategy = cluster.Spec.Rke2Config.ClusterUpgradeStrategy
-
-	}
-	if updateVersion == "" {
```

Comment:
Q: Why do we drop this check?

Reply:
Probably, I realize now it's also supposed to be checking if `cluster.Spec.Rke2Config.Version == ""`, and not just the cluster driver. Good catch!